Cython/Compiler/ExprNodes.py

```diff
@@ -14115,20 +14120,15 @@ def analyse_types(self, env):
 
         self.rhs = self.rhs.analyse_types(env)
         if not self.rhs.arg.is_temp:
-            if not self.rhs.arg.is_literal:
+            if self.rhs.arg.is_simple():
+                # For literals we can optimize by just using the literal twice
+                self.assignment.rhs = copy.copy(self.rhs.arg)
```

Comment:
I'm not sure if `is_simple()` only applies to nodes for which shallow copying is safe. I'm not even sure that it's entirely safe for literals. What about constant tuples? Those are considered literals, but shallow copying them would probably remove the link to the global C constant that holds them.